src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/support/StaffEmailSupportController.java

```diff
@@ -0,0 +1,240 @@
+        String fullName = userDetails.getFullName();
+
+        if (!claim.getClaimData().isClaimantRepresented()) {
+            GeneratePinResponse pinResponse = userService.generatePin(
+                claim.getClaimData().getDefendant().getName(),
+                authorisation
+            );
+
+            claimService.linkLetterHolder(claim, pinResponse.getUserId(), authorisation);
+            documentGenerator.generateForNonRepresentedClaim(
+                new CitizenClaimIssuedEvent(claim, pinResponse.getPin(), fullName, authorisation)
+            );
+        } else {
+            documentGenerator.generateForRepresentedClaim(
+                new RepresentedClaimIssuedEvent(claim, fullName, authorisation)
+            );
+        }
+    }
+
+    private void resendStaffNotificationForIntentToProceed(Claim claim, String authorization) {
+        ClaimantResponse claimantResponse = claim.getClaimantResponse()
+            .orElseThrow(() -> new IllegalArgumentException(MISSING_CLAIMANT_RESPONSE));
+
+        if (claimantResponse.getType() != REJECTION) {
+            throw new IllegalArgumentException("Rejected Claimant Response is mandatory for 'intent-to-proceed' event");
+        }
+
+        claimantResponseStaffNotificationHandler.notifyStaffWithClaimantsIntentionToProceed(
+            new ClaimantResponseEvent(claim, authorization)
+        );
+    }
+
+    private void resendStaffNotificationOnMoreTimeRequested(Claim claim) {
+        if (!claim.isMoreTimeRequested()) {
+            throw new ConflictException("More time has not been requested yet - cannot send notification");
+        }
+
+        // Defendant email is not available at this point however it is not used in staff notifications
+        MoreTimeRequestedEvent event = new MoreTimeRequestedEvent(claim, claim.getResponseDeadline(), null);
+        moreTimeRequestedStaffNotificationHandler.sendNotifications(event);
+    }
+
+    private void resendStaffNotificationOnDefendantResponseSubmitted(Claim claim, String authorization) {
+        if (!claim.getResponse().isPresent()) {
+            throw new ConflictException(CLAIM + claim.getReferenceNumber() + " does not have associated response");
+        }
+        DefendantResponseEvent event = new DefendantResponseEvent(claim, authorization);
+        defendantResponseStaffNotificationHandler.onDefendantResponseSubmitted(event);
+    }
+
+    private void resendStaffNotificationOnAgreementCountersigned(Claim claim, String authorisation) {
+        if (claim.getSettlementReachedAt() == null) {
+            throw new ConflictException(CLAIM + claim.getReferenceNumber() + " does not have a settlement");
+        }
+        AgreementCountersignedEvent event = new AgreementCountersignedEvent(claim, null, authorisation);
+        agreementCountersignedStaffNotificationHandler.onAgreementCountersigned(event);
+    }
+
+    private void resendStaffNotificationClaimantResponse(Claim claim, String authorization) {
```

Comment:
resendStaffNotificationClaimantResponse calls claimantResponseStaffNotificationHandler. Within the handler based upon the logic we call one of the four different services which means we are excluding the whole logic within the handler which is not in the scope of staff email decommission  stories.
Out of scope:
·     ccjStaffNotificationHandler.onInterlocutoryJudgmentEvent
·     rejectOrgPaymentPlanStaffNotificationService.notifyStaffClaimantRejectOrganisationPaymentPlan